tools/network/cloudflare/cloudflare.go

```diff
@@ -295,6 +301,37 @@ func (c *Cred) GetZones(ctx context.Context) (zones []Zone, err error) {
 	return zones, err
 }
 
+// GetZoneID returns a zoneID that matches the requested zoneDomainName.
+func (c *Cred) GetZoneID(ctx context.Context, zoneDomainName string) (zoneID string, err error) {
+	zoneID = ""
+	zones, err := c.GetZones(ctx)
+	if err != nil {
+		return
+	}
+	if len(zones) == 0 {
+		err = ErrUserNotPermitted
```

Comment:
Is no zones really a permissions issue?  Seems like err would be non-nil in that case.

Reply:
A user account might have no access on any of the zones. In that case, the authentication is valid, but the user cannot act on any of the zones.
For the scope of this function ( where we want to find the zoneID ), the user is not permitted on that zone.